examples/custom_rule.jl

```diff
@@ -0,0 +1,262 @@
+# * We annotated `x` and `y` with [`Duplicated`](@ref), similar to our first simple forward rule.
+
+# Now, let's unpack the body of our `augmented_primal` rule:
+# * We checked if the `config` requires the primal. If not, we need not compute the return value, but we make sure to mutate `y` in all cases.
+# * We checked if `x` could possibly be overwritten using the `Overwritten` attribute of [`EnzymeRules.Config`](@ref). 
+#   If so, we save the value of `x` on the `tape` of the returned [`EnzymeRules.AugmentedReturn`](@ref) objet.
+# * We return a shadow of `nothing` since the return value is [`Active`](@ref) and hence does not need a shadow.
+
+# Now, we write a method for [`EnzymeRules.reverse`](@ref):
+
+function reverse(config::ConfigWidth{1}, func::Const{typeof(f)}, dret::Active, tape,
+                 y::Duplicated, x::Duplicated)
+    println("In custom reverse rule.")
+    ## retrieve x value, either from original x or from tape if x may have been overwritten.
+    xval = overwritten(config)[3] ? tape : x.val 
+    ## accumulate into x's shadow, don't assign!
+    x.dval .+= 2 .* xval .* dret.val 
+    return ()
+end
+
+# The activities used in the signature match what we used for `augmented_primal`. 
+# One key difference is that we now receive an *instance* `dret` of [`Active`](@ref) for the return type, not just a type annotation.
+# Here, `dret.val` stores the derivative value for `dret` (not the original return value!).
+# Using this derivative value, we accumulate the backpropagated derivatives for `x` into its shadow. 
+# Note that we do not accumulate anything into `y`'s shadow! This is because `y` is overwritten within `f`, so there is no derivative
+# w.r.t. to the `y` that was originally inputted.
+
+# Finally, let's see our reverse rule in action!
+
+x  = [3.0, 1.0]
+dx = [0.0, 0.0]
+y  = [0.0, 0.0]
+dy = [0.0, 0.0]
+
+g(y, x) = f(y, x)^2
+
+autodiff(Reverse, g, Duplicated(y, dy), Duplicated(x, dx))
+@show dx # derivative of g w.r.t. x
+@show dy; # derivative of g w.r.t. y
+
+# Let's also try a function which mutates `x` after running `f`:
+
+function h(y, x)
+    ret = f(y, x)
+    x .= x.^2
+    return ret^2
+end
+
+x  = [3.0, 1.0]
+y  = [0.0, 0.0]
+dx .= 0
+dy .= 0
+
+autodiff(Reverse, h, Duplicated(y, dy), Duplicated(x, dx))
+@show dx # derivative of h w.r.t. x
+@show dy; # derivative of h w.r.t. y
+
+# ## Marking functions inactive
+
+# If we want to tell Enzyme that the function call does not affect the differentiation result in any form 
+# (i.e. not by side effects or through its return values), we can simply use [`EnzymeRules.inactive`](@ref).
+# So long as there exists a matching dispatch to [`EnzymeRules.inactive`](@ref), the function will be considered inactive.
+# For example:
+
+function printhi()
+    println("Hi!")
+end
+
+EnzymeRules.inactive(::typeof(printhi), args...) = nothing
+
```

Comment:
Maybe add another example like the following

myidentity(x) = x

inactive(typeof(x)) = ...

function tester(x)
  return x + myidentity(x)
 end
 
 ^ the above should return a derivative of 1.0 not 2.0